lib/puppetserver/ca/action/create.rb

```diff
@@ -141,6 +152,10 @@ def generate_key_csr(certname, settings, digest)
           private_key = host.create_private_key(settings[:keylength])
           csr = host.create_csr(certname, private_key)
 
+          if !settings[:subject_alt_names].empty?
+            host.add_csr_extension(csr, "subjectAltName", settings[:subject_alt_names])
+          end
+
```

Comment:
So I tried this out, and it doesn't work, I think you need to add the extension before you sign the CSR. So maybe pass the names into `create_csr` and add the extension in there?

Reply:
^ Yep, that would make sense.  Thanks Maggie